src/interface_py/h2o4gpu/util/gpu.py

```diff
@@ -54,15 +56,20 @@ def get_gpu_info(return_usage=False):
     res = None
     # sometimes hit broken process pool in cpu mode,
     # so just return back no gpus.
-    try:
-        with ProcessPoolExecutor(max_workers=1) as executor:
-            future = executor.submit(get_gpu_info_subprocess, return_usage)
-            res = future.result()
-        return res
-    except concurrent.futures.process.BrokenProcessPool:
-        if return_usage:
-            return (total_gpus, total_mem, gpu_type, usage)
-        return (total_gpus, total_mem, gpu_type)
+    for trial in range(0, trials):
+        try:
+            with ProcessPoolExecutor(max_workers=1) as executor:
+                future = executor.submit(get_gpu_info_subprocess, return_usage)
+                # don't wait more than 30s, import on py3nvml can hang if 2 subprocesses GIL lock import at same time
+                res = future.result(timeout=timeout)
```

Comment:
I've been wondering why are we doing this with a subprocess? Can't we just call `get_gpu_info_subprocess` since we call `result` (which I assume is synchronous anyway) but we expose ourselves to potential deadlocks/race conditions? Wouldn't that fix the problem?

Or at least we can add a comment why are we doing it this way.